### PR TITLE
Regenerate with latest gir/gir-files

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: glib/src/auto/functions.rs [hunk 1]
```diff
@@ -701,25 +701,22 @@ pub fn spaced_primes_closest(num: u32) -> u32 {
...
 
 #[doc(alias = "g_spawn_async")]
-pub fn spawn_async(
+pub fn spawn_async<P: FnOnce() + Send + Sync + 'static>(
     working_directory: Option<impl AsRef<std::path::Path>>,
     argv: &[&std::path::Path],
     envp: &[&std::path::Path],
     flags: SpawnFlags,
-    child_setup: Option<Box_<dyn FnOnce() + 'static>>,
+    child_setup: P,
```
Comment (sdroege) [inline glib/src/auto/functions.rs:709]: This is wrong. The callback is optional :)

Comment (sdroege) [reply glib/src/auto/functions.rs:709]: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3523

Comment (bilelmoussaoui) [reply glib/src/auto/functions.rs:709]: This change still didn't land in the gir-file...

Comment (sdroege) [reply glib/src/auto/functions.rs:709]: Need to send an MR to gobject-introspection to update its glib snapshot

**Comment by sdroege** [comment]
Looks good to me, just need to get gobject-introspection fixed.

File: glib/src/auto/functions.rs (hunk 1 above)
Comment (sdroege) [reply glib/src/auto/functions.rs:709]: https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/420

Comment (bilelmoussaoui) [reply glib/src/auto/functions.rs]: Fixed

**Comment by sdroege** [comment]
@bilelmoussaoui This is ready now?

**Comment by bilelmoussaoui** [comment]
Yes, should be all good

**Comment by sdroege** [comment]
Nice, will review some time later today :)

**Comment by sdroege** [comment]
This is marked `needs-backport` but contains breaking changes (dropping of the `Display` impls at least). Needs special care when backporting to only include the OK changes.

File: cairo/src/xcb.rs [hunk 2]
```diff
@@ -19,12 +19,6 @@ impl XCBDrawable {
     }
 }
 
-impl fmt::Display for XCBDrawable {
```
Comment (sdroege) [inline cairo/src/xcb.rs:22]: `unused import: `fmt``

File: gio/src/win32_input_stream.rs [hunk 3]
```diff
@@ -33,12 +33,6 @@ impl<O: IsA<Win32InputStream>> Win32InputStreamExt for O {
     }
 }
 
-impl fmt::Display for Win32InputStream {
```
Comment (sdroege) [inline gio/src/win32_input_stream.rs:36]: `unused import: `fmt``

File: gio/src/win32_output_stream.rs [hunk 4]
```diff
@@ -31,12 +31,6 @@ pub trait Win32OutputStreamExt: IsA<Win32OutputStream> + 'static {
 
 impl<O: IsA<Win32OutputStream>> Win32OutputStreamExt for O {}
 
-impl fmt::Display for Win32OutputStream {
```
Comment (sdroege) [inline gio/src/win32_output_stream.rs:34]: `unused import: `fmt``

File: glib/src/collections/list.rs [hunk 5]
```diff
@@ -658,7 +658,7 @@ pub struct IterMut<'a, T: TransparentPtrType> {
 
 impl<'a, T: TransparentPtrType> IterMut<'a, T> {
     #[inline]
-    fn new(list: &'a mut List<T>) -> IterMut<'a, T> {
```
Comment (sdroege) [inline glib/src/collections/list.rs:661]: This is wrong. The iterator gives mutable access to the list so we need a mutable reference

File: glib/src/collections/slist.rs [hunk 6]
```diff
@@ -652,7 +652,7 @@ pub struct IterMut<'a, T: TransparentPtrType> {
 
 impl<'a, T: TransparentPtrType> IterMut<'a, T> {
     #[inline]
-    fn new(list: &'a mut SList<T>) -> IterMut<'a, T> {
```
Comment (sdroege) [inline glib/src/collections/slist.rs:655]: This is wrong. The iterator gives mutable access to the list so we need a mutable reference

File: cairo/src/context.rs [hunk 7]
```diff
@@ -53,12 +53,6 @@ impl fmt::Debug for RectangleList {
     }
 }
 
-impl fmt::Display for RectangleList {
```
Comment (sdroege) [inline cairo/src/context.rs:56]: `win32_surface.rs` has an unused `fmt` import

**Comment by sdroege** [comment]
I'll update this and then merge

File: glib/src/collections/list.rs (hunk 5 above)
Comment (sdroege) [reply glib/src/collections/list.rs]: Apparently clippy also does not warn about this anymore.

**Comment by sdroege** [comment]
@bilelmoussaoui can you make a backport PR of this with the things that should be included?

**Comment by bilelmoussaoui** [comment]
@sdroege i don't think there is much to backport actually, the time i put the label i thought remove display was not a breaking change :P

**Comment by sdroege** [comment]
Yeah I was wondering which changes you wanted to backport. Let's not do that then :)

**Pull request merged**